tests/integration/test_charm.py

```diff
@@ -39,59 +49,59 @@ async def test_deploy(ops_test: OpsTest, charm: str, series: str):
 
-    # Issuing dummy update_status just to trigger an event.
+    # Reducing the update status frequency to speed up the triggering of deferred events.
     await ops_test.model.set_config({"update-status-hook-interval": "10s"})
 
     await ops_test.model.wait_for_idle(apps=[application_name], status="active", timeout=1000)
     assert ops_test.model.applications[application_name].units[0].workload_status == "active"
 
-    # Effectively disable the update status from firing.
-    await ops_test.model.set_config({"update-status-hook-interval": "60m"})
-
 
 @pytest.mark.abort_on_fail
 @pytest.mark.parametrize("series", SERIES)
-async def test_database_is_up(ops_test: OpsTest, series: str):
+@pytest.mark.parametrize("unit_id", UNIT_IDS)
+async def test_database_is_up(ops_test: OpsTest, series: str, unit_id: int):
     # Set a composite application name in order to test in more than one series at the same time.
     application_name = build_application_name(series)
 
     # Query Patroni REST API and check the status that indicates
     # both Patroni and PostgreSQL are up and running.
-    host = ops_test.model.units.get(f"{application_name}/0").public_address
+    host = await get_unit_address(ops_test, f"{application_name}/{unit_id}")
     result = requests.get(f"http://{host}:8008/health")
     assert result.status_code == 200
 
 
 @pytest.mark.parametrize("series", SERIES)
-async def test_settings_are_correct(ops_test: OpsTest, series: str):
+@pytest.mark.parametrize("unit_id", UNIT_IDS)
+async def test_settings_are_correct(ops_test: OpsTest, series: str, unit_id: int):
     # Connect to the PostgreSQL instance.
     # Set a composite application name in order to test in more than one series at the same time.
     application_name = build_application_name(series)
 
     # Retrieving the postgres user password using the action.
-    action = await ops_test.model.units.get(f"{application_name}/0").run_action(
+    action = await ops_test.model.units.get(f"{application_name}/{unit_id}").run_action(
         "get-initial-password"
     )
     action = await action.wait()
     password = action.results["postgres-password"]
 
     # Connect to PostgreSQL.
-    host = ops_test.model.units.get(f"{application_name}/0").public_address
+    host = await get_unit_address(ops_test, f"{application_name}/{unit_id}")
     logger.info("connecting to the database host: %s", host)
     with psycopg2.connect(
         f"dbname='postgres' user='postgres' host='{host}' password='{password}' connect_timeout=1"
-    ) as connection, connection.cursor() as cursor:
+    ) as connection:
         assert connection.status == psycopg2.extensions.STATUS_READY
 
         # Retrieve settings from PostgreSQL pg_settings table.
         # Here the SQL query gets a key-value pair composed by the name of the setting
         # and its value, filtering the retrieved data to return only the settings
         # that were set by Patroni.
+        cursor = connection.cursor()
```

Comment:
This could probably be closed after line 111

Reply:
I wonder if cursors are closed automagically when the connection is closed (when exiting the context manager)

Reply:
Good catch! Cursors are closed when exiting the context manager, but I figured out that connection dont. SoI fixed it using the syntax explained in https://github.com/psycopg/psycopg2/blob/f469331af5f7b236f8037b302b69c2bf0df2a096/doc/src/connection.rst (keeping the connection in different context managers to have scope for transactions, and closing the connection later).